src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanNoArgConstructorQuickFix.java

```diff
@@ -37,57 +43,65 @@
+
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        final PsiElement node = context.getCoveredNode();
+        final PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
+
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
+            String constructorName = toResolve.getTitle();
+            ChangeCorrectionProposal proposal = new AddConstructorProposal(constructorName,
+                    context.getSource().getCompilationUnit(), context.getASTRoot(), parentType, 0,
+                    constructorName.equals("AddProtectedConstructor") ? "protected" : "public");
 
-            codeActions.addAll(addConstructor(diagnostic, context, parentType));
+            try {
+                WorkspaceEdit we = context.convertToWorkspaceEdit(proposal);
+                toResolve.setEdit(we);
+            } catch (Exception e) {
+                LOGGER.log(Level.WARNING, "Unable to create workspace edit for code actions to add constructors", e);
+            }
 
-            return codeActions;
         }
-        return Collections.emptyList();
+        return toResolve;
     }
 
     protected PsiClass getBinding(PsiElement node) {
         return PsiTreeUtil.getParentOfType(node, PsiClass.class);
     }
 
-    private List<CodeAction> addConstructor(Diagnostic diagnostic, JavaCodeActionContext context, PsiClass notUsed) {
+    private List<CodeAction> addConstructor(Diagnostic diagnostic, JavaCodeActionContext context) {
         List<CodeAction> codeActions = new ArrayList<>();
-        JavaCodeActionContext targetContext = null;
-        PsiElement node = null;
-        PsiClass parentType = null;
-
-        // option for protected constructor
-        targetContext = context.copy();
-        node = targetContext.getCoveredNode();
-        parentType = getBinding(node);
-        String name = Messages.getMessage("AddProtectedConstructor");
-        ChangeCorrectionProposal proposal = new AddConstructorProposal(name,
-                targetContext.getSource().getCompilationUnit(), targetContext.getASTRoot(), parentType, 0);
-        CodeAction codeAction = targetContext.convertToCodeAction(proposal, diagnostic);
-
-        if (codeAction != null) {
-            codeActions.add(codeAction);
-        }
-
-        // option for public constructor
-        targetContext = context.copy();
-        node = targetContext.getCoveredNode();
-        parentType = getBinding(node);
-        name = Messages.getMessage("AddPublicConstructor");
-        proposal = new AddConstructorProposal(name,
-                targetContext.getSource().getCompilationUnit(), targetContext.getASTRoot(), parentType, 0, "public");
-        codeAction = targetContext.convertToCodeAction(proposal, diagnostic);
+        String[] constructorNames = {"AddProtectedConstructor", "AddPublicConstructor"};
```

Comment:
In previous code name is setting from Message.properties.
String name = Messages.getMessage("AddProtectedConstructor");
Please change use the same.
```suggestion
        String[] constructorNames = {Messages.getMessage("AddProtectedConstructor"), Messages.getMessage("AddPublicConstructor")};
```
Similarly make changes in line no. 67